analyses/cell-type-wilms-tumor-06/notebook_template/06_cnv_infercnv_exploration.Rmd

```diff
@@ -0,0 +1,445 @@
+
+- `group.by` is the metadata used for grouping the violin plots 
+
+```{r }
+visualize_feature <- function(object, features, group.by ){
+ 
+                  d <- SCpubr::do_FeaturePlot(object, 
+                                              features = features, 
+                                              pt.size = 0.2, 
+                                              legend.width = 0.5, 
+                                              legend.length = 5, 
+                                              legend.position = "right") + ggtitle(features)
+                  b <- SCpubr::do_ViolinPlot(object, 
+                                             features = features, 
+                                             ncol = 1, 
+                                             group.by = group.by, 
+                                             legend.position = "none",
+                                             assay = "SCT") + ylab(features)
+                   
+                  return(d + b + plot_layout(ncol = 2, widths = c(2,4))) 
+}
+```
+
+
+#### Visualize CNV density
+
+For a Seurat object `object`and a features `features`, the function `visualize_feature` will plot `FeaturePlot` and `ViolinPlot`
+
+- `object` is the Seurat object 
+
+- `features` the gene or quantitative value to be plotted
+
+- `group.by` is the metadata used for grouping the violin plots 
+
+```{r }
+visualize_density <- function(object, features, group.by ){
+ 
+                  d <- SCpubr::do_RidgePlot(object, 
+                                            feature = features, 
+                                            group.by = group.by, 
+                                             legend.position = "none",
+                                             assay = "SCT") + ylab(features)
+                   
+                  return(d) 
+}
+          
+ 
+```
+
+### Base directories
+
+```{r base paths, eval=TRUE}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
+```
+
+
+### Input files
+
+The input for this notebook are the results of `06_inferCNV.R`
+
+```{r}
+
+result_dir <- file.path(module_base, "results", params$sample_id, "06_infercnv")
+
+```
+
+#### `Seurat` object
+
+We load the `Seurat` object generated in `06_infercnv.R`
+
+```{r}
+
+srat <- readRDS(file.path(module_base, "results", params$sample_id, glue::glue("02b-fetal_kidney_label-transfer_", params$sample_id, ".Rds")))
+
+```
+
+## Analysis
+
+### Heatmap of infercnv results
+
+Here we plot the output of `infercnv` as heatmaps of CNV.
+We first look at the png file generated by the `infercnv` function.
+We then used the `infercnv object` to look at mean CNV value across compartments (immune, endothelial, stroma and fetal nephron). 
```

Comment:
At first I was going to leave a review here that it's worth noting these results are generated without HMM. But then I wondered - are these results identical regardless of HMM, and specifying an HMM just gives you _additional_ information on top of what running without would give you? Is that your experience with infercnv? If yes, then this simplifies how you can run `06_infercnv.R`. You would never need to run it without an HMM, since that would just duplicate the main CNV results.  If no, and results are different, then why don't we show the heatmaps here for HMM results?

Reply:
From my understanding, HMM prediction is an _additional_ step happening _after_ the [standard inferCNV processing routines](https://github.com/broadinstitute/infercnv/wiki/Running-InferCNV). 
I manually checked the heatmaps `06_infercnv_{sample_id}_reference-both_HMM-no_heatmap.png` and `06_infercnv_{sample_id}_reference-both_HMM-i3_heatmap.png`, they look indeed identical. 

The reason why I first ran `infercnv` without the additional HMM CNV prediction model was to _quickly_ compare the choice of the _normal reference_. I think the HMM model is not required to look at the variability between the CNV profiles when taking no, or immune and/or endothelial cells as normal reference. 

From thee 4 heatmaps, I conclude that **we need to use normal cells as reference**. Else, the mean of expression is taking as the normal and the more abundant cells will thus look normal, as illustrate with the example in sample -179:
![image](https://github.com/user-attachments/assets/ec913c1c-63c9-47e4-bcc4-204a08b07390)
Here, fetal nephron, which we _know/think_ contains cancer cells, looks the more _normal_. Also chr17 seems to be gained in immune and endothelial cells, which is an artefact due to the fact that chr17 encodes lot of chemokines secreted by the 2 cells types. 

Then, taking the endothelial cells as reference, it seems that _all_ compartments have a chr3 and chr5 loss. 
![image](https://github.com/user-attachments/assets/f006f1f7-2c42-4016-a4ab-3ac05da4822f)

This is however an artefact due to the fact that key endothelial genes are encoded in chr3 and chr5. This artefact is corrected when using both immune and endothelial cells in the reference. 
![image](https://github.com/user-attachments/assets/ac37eb10-906c-42dd-81a0-ddf99e558796)

Reply:
> you would never need to run it without an HMM, since that would just duplicate the main CNV results.

I agree for the condition `-- reference "both"`, I could only run it with the HMM model. 
For the other `--reference` conditions, I would however not spend too much time and resources calculating the HMM model.  

Is that what you mean?

Reply:
> The reason why I first ran infercnv without the additional HMM CNV prediction model was to quickly compare the choice of the normal reference.

This makes total sense to me as the right place to start! 

> Is that what you mean?

Yes, this basically what I mean! At this point, we know that the results are the same with and without HMM. I also looked a bit through inferCNV's code, and it indeed seems that the HMM doesn't affect the other code. There is just an `if` statement to add on the HMM, but it doesn't change the other calculations. Therefore, you never need to run without HMM. Even if this only saves a small amount of time and resources, especially for running in CI.